### DISCUSSION TITLE
custom Server  Router.push jump page loading is very slow, I am very anxious to solve this problem!thank you

### DISCUSSION THREAD

**Discussion by parlay96** [vercel/next.js#30834] [Help]
### What version of Next.js are you using?

 10.0.1

### What version of Node.js are you using?

12.0.0

### What browser are you using?

Chrome

### What operating system are you using?

Windows

### How are you deploying your application?

 next start

### Describe the Bug

![image](https://user-images.githubusercontent.com/32978177/139836409-2e141a85-9673-4716-82ac-7e7d614de831.png)
![image](https://user-images.githubusercontent.com/32978177/139836515-e3798cc7-30f0-4471-af83-a323376f9e4a.png)
I just want to implement a configurable path. For example, my page level is very deep, and then I want to customize the server
Why does the custom server load slowly?

### Expected Behavior

——pages
         —textModule
                -a.tsx
       —b.tsx
At this time, the access path is：  www.abc.com/textModule/a,  But I want to access the path:   www.abc.com/a,

This is the deficiency of the next routing method,   I thought of a custom server to solve my path access problem. 
However, there will be a problem that the jump page is very slow

### To Reproduce

——pages
         —textModule
                -a.tsx
       —b.tsx
At this time, the access path is：  www.abc.com/textModule/a,  But I want to access the path:   www.abc.com/a,

This is the deficiency of the next routing method,   I thought of a custom server to solve my path access problem. 
However, there will be a problem that the jump page is very slow

**Comment by shuding** [comment]
Hey! Please keep in mind that we can't help with your own application issues here, if you think there is a bug **in Next.js** (not your application code), please open a proper issue with detailed reproduction that we can check, such as a repository we can clone.

(We can also chat in Chinese if that's easier! Also feel free to reach out to me via Telegram: sding or WeChat: dingshu27.)

**Comment by parlay96** [comment]
@shuding I added your wechat
